handlers/categories.go

```diff
@@ -0,0 +1,116 @@
+package handlers
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"path"
+
+	"github.com/Daniorocket/cookit-backend/models"
+	"github.com/gorilla/mux"
+	uuid "github.com/satori/go.uuid"
+)
+
+func (d *Handler) GetListOfCategories(w http.ResponseWriter, r *http.Request) {
+	page := r.URL.Query().Get("page")
+	limit := r.URL.Query().Get("limit")
+	categories, te, err := models.GetAllCategories(d.Client, d.DatabaseName, page, limit)
+	if err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Failed to get list of categories:"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		return
+	}
+	if err := CreateApiResponse(w, Pagination{
+		Data:          categories,
+		Limit:         limit,
+		Page:          page,
+		TotalElements: te,
+	}, http.StatusOK, "success", "none"); err != nil {
+		log.Println("Error create Api response:", err)
+	}
+}
+func (d *Handler) CreateCategory(w http.ResponseWriter, r *http.Request) {
+	mr, err := r.MultipartReader()
+	if err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusInternalServerError, "failed", "Failed to load image:"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		return
+	}
+
+	cat := models.Category{}
+	for {
```

Comment:
Tego nie da się gdzieś wynieść do funkcji żeby było reużywalne?

Reply:
Czy działanie tej funkcji ma w tym wypadku zwrócić pustą kategorię (lub równoważnie przepis, użytkownika) przy inicjalizacji projektu, a następnie ewentualnie zamiast tworzyć nowe obiekty danego typu, to modyfikuje istniejący obiekt, który zostanie zwrócony przez tą funkcję?

Reply:
Może Ci zwrócić ona jsona do zdekodowania i plik który znalazła np.
```go
json, file, err := DecodeMultipartRequest(r)
```

Reply:
Już rozumiem, myślałem, że chodzi konkretnie o linię 47, a tu o całą funkcję, racja, dziękuję